src/test/java/com/brandwatch/internship/fetchandstoretweets/integrationtests/QueryIT.java

```diff
@@ -0,0 +1,66 @@
+package com.brandwatch.internship.fetchandstoretweets.integrationtests;
+
+import com.brandwatch.internship.fetchandstoretweets.controllers.QueryController;
+import com.brandwatch.internship.fetchandstoretweets.entities.Query;
+import com.brandwatch.internship.fetchandstoretweets.repositories.QueryRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@TestPropertySource(locations="classpath:test.properties")
+public class QueryIT {
+
+    private static final long QUERY_ID = 3;
+    private static final String SEARCH_STRING = "test";
+
+    @Autowired
+    private QueryController queryController;
+
+    @Autowired
+    private QueryRepository queryRepository;
+
+
+    @Test
+    public void getAllQueriesTest() {
+        assertEquals(queryController.getAllQueries(), queryRepository.findAll());
```

Comment:
I think it's worth mentioning that you are calling controller methods directly in all your ITs, but also could easily make actual HTTP requests to the endpoints instead, which would be "even more" end-to-end.  If this is what you've discussed with Anton that's fine.

Reply:
Yeah I know. I made it like that because I can not see a big difference between these 2 ways. I think the difference is that the http requests also the url to see if there is a typo or wrong port. Am I right? should I change it?

Reply:
The difference could also be serialisation and deserialisation of objects, which doesn't need to happen if you call the method directly.

Reply:
Oh true. I somehow assumed that you send the search string as part of the body, but you don't.